standalone-metastore/metastore-tools/metastore-benchmarks/src/main/java/org/apache/hadoop/hive/metastore/tools/ACIDBenchmarks.java

```diff
@@ -0,0 +1,247 @@
+  }
+
+  @State(Scope.Benchmark)
+  public static class TestLocking extends CoreContext {
+    private int nTables;
+
+    @Param("0")
+    private int nPartitions;
+
+    private List<LockComponent> lockComponents;
+
+    @Setup
+    public void setup() {
+      this.nTables = (nPartitions != 0) ? howMany / nPartitions : howMany;
+      createLockComponents();
+    }
+
+    @State(Scope.Thread)
+    public static class ThreadState extends CoreContext.ThreadState {
+      List<Long> openTxns = new ArrayList<>();
+      long txnId;
+
+      @Setup(org.openjdk.jmh.annotations.Level.Invocation)
+      public void iterSetup() {
+        txnId = executeOpenTxnAndGetTxnId(client);
+        LOG.debug("opened txn, id={}", txnId);
+        openTxns.add(txnId);
+      }
+
+      @TearDown
+      public void doTearDown() throws Exception {
+        client.abortTxns(openTxns);
+        if (BenchmarkUtils.checkTxnsCleaned(client, openTxns) == false) {
+          LOG.error("Something went wrong with the cleanup of txns");
+        }
+        LOG.debug("aborted all opened txns");
+      }
+    }
+
+    @Benchmark
+    public void lock(TestLocking.ThreadState state) {
+      LOG.debug("sending lock request");
+      executeLock(state.client, state.txnId, lockComponents);
+    }
+
+    private void createLockComponents() {
+      lockComponents = new ArrayList<>();
+
+      for (int i = 0; i < nTables; i++) {
+        for (int j = 0; j < nPartitions - (nPartitions > 1 ? 1 : 0); j++) {
+          lockComponents.add(
+            new Util.LockComponentBuilder()
+              .setDbName("default")
+              .setTableName(String.format("tmp_table_%d", i))
+              .setPartitionName("p_" + j)
+              .setShared()
+              .setOperationType(DataOperationType.SELECT)
+              .build());
+        }
+        if (nPartitions != 1) {
+          lockComponents.add(
+            new Util.LockComponentBuilder()
+              .setDbName("default")
+              .setTableName(String.format("tmp_table_%d", i))
+              .setShared()
+              .setOperationType(DataOperationType.SELECT)
+              .build());
+        }
+      }
+    }
+
+    private static long executeOpenTxnAndGetTxnId(HMSClient client) {
+      return throwingSupplierWrapper(() -> client.openTxn(1).get(0));
+    }
+
+    private void executeLock(HMSClient client, long txnId, List<LockComponent> lockComponents) {
+      LockRequest req = new LockRequest(lockComponents, "hclient", "localhost");
+      req.setTxnid(txnId);
+      throwingSupplierWrapper(() -> client.lock(req));
+    }
+  }
+
+  @State(Scope.Benchmark)
+  public static class TestAllocateTableWriteIds extends CoreContext {
+    String dbName = "test_db";
+    String tblName = "tmp_table";
+
+    @State(Scope.Thread)
+    public static class ThreadState extends CoreContext.ThreadState {
+      List<Long> openTxns = new ArrayList<>();
+      long txnId;
+
+      @Setup
+      public void iterSetup() {
```

Comment:
rename to `doSetup` as it's global and not iteration scoped

Reply:
renaming to simple setup, as doSetup would overwrite the parent doSetup()

Reply:
i think, it will execute both, check for example doTearDown in ThreadState confs.